lib/reek/report.rb

```diff
@@ -0,0 +1,62 @@
+require_relative 'report/report'
+require_relative 'report/formatter'
+require_relative 'report/heading_formatter'
+
+module Reek
+  # Reek reporting functionality.
+  module Report
+    # Make report format symbol to a report class. Unknown symbols are mapped to +TextReport+.
+    #
+    # @param [Symbol] report_format The format to map
+    #
+    # @returns The mapped report class
+    #
+    # @api private
+    def self.report_class(report_format)
+      case report_format
+      when :yaml
+        YAMLReport
+      when :json
+        JSONReport
+      when :html
+        HTMLReport
+      when :xml
+        XMLReport
+      else # :text
+        TextReport
+      end
+    end
+
+    # @api private
+    def self.location_formatter(location_format)
+      case location_format
+      when :single_line
+        SingleLineLocationFormatter
+      when :plain
+        BlankLocationFormatter
+      else # :numbers
+        DefaultLocationFormatter
+      end
+    end
+
+    # @api private
+    def self.heading_formatter(heading_format)
+      case heading_format
+      when :verbose
+        HeadingFormatter::Verbose
+      else # :quiet
+        HeadingFormatter::Quiet
+      end
+    end
+
+    # @api private
+    def self.warning_formatter_class(warning_format)
+      case warning_format
+      when :wiki_links
+        WikiLinkWarningFormatter
+      else
+        SimpleWarningFormatter
+      end
+    end
+  end
```

Comment:
Can we turn them all into `module_functions`? They’d be callable on `Report` but private if some class decides to `include Report`.